### PR TITLE
Feature/10317 port request (authorize otp els)

### PR DESCRIPTION
**Pull request by 30mar**
## Description
Porting the otp authorization for ELS from the old HTTP Client to the new rest Service.

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-10317

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/HTTPClient/Resources/OTPAuthorizationForELS/OTPResponsePropertiesReceiveModel.swift [hunk 1]
```diff
@@ -0,0 +1,10 @@
+//
+// 🦠 Corona-Warn-App
+//
+
+import Foundation
+
+struct OTPResponsePropertiesReceiveModel: Codable {
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/HTTPClient/Resources/OTPAuthorizationForELS/OTPResponsePropertiesReceiveModel.swift:7]: Shouldn't it be just OTPReceiveModel

File: src/xcode/ENA/ENA/Source/HTTPClient/Resources/OTPAuthorizationForELS/OTPAuthorizationForELSResource.swift [hunk 2]
```diff
@@ -0,0 +1,163 @@
+//
+// 🦠 Corona-Warn-App
+//
+
+import Foundation
+import jsonfunctions
+enum OTPAuthorizationELSError: LocalizedError, Equatable {
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/HTTPClient/Resources/OTPAuthorizationForELS/OTPAuthorizationForELSResource.swift:7]: OTPAuthorizationForELSError ?

File: src/xcode/ENA/ENA/Source/HTTPClient/Resources/OTPAuthorizationForELS/OTPAuthorizationForELSResource.swift [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+			default:
+				Log.error("Failed to authorize OTP - response error", log: .api)
+				Log.error(String(statusCode), log: .api)
+				return .internalServerError
+			}
+		default:
+			return .invalidResponseError
+		}
+	}
+	private func otpAuthorizationFailureHandler(for response: Data?, statusCode: Int) -> OTPAuthorizationELSError? {
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/HTTPClient/Resources/OTPAuthorizationForELS/OTPAuthorizationForELSResource.swift:123]: Mark missing

File: src/xcode/ENA/ENA/Source/HTTPClient/Resources/OTPAuthorizationForELS/OTPAuthorizationForELSResource.swift [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+
+	static func == (lhs: OTPAuthorizationELSError, rhs: OTPAuthorizationELSError) -> Bool {
+		return lhs.description == rhs.description
+	}
+
+}
+
+struct OTPAuthorizationForELSResource: Resource {
+	
+	init(
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/HTTPClient/Resources/OTPAuthorizationForELS/OTPAuthorizationForELSResource.swift:63]: Mark missing

File: src/xcode/ENA/ENA/Source/HTTPClient/Resources/OTPAuthorizationForELS/OTPResponsePropertiesReceiveModel.swift (hunk 1 above)
Comment (30mar) [reply src/xcode/ENA/ENA/Source/HTTPClient/Resources/OTPAuthorizationForELS/OTPResponsePropertiesReceiveModel.swift:7]: there is another OTP, EDUS, so i wanted to be specific

**Review by puneetmahali1 — APPROVED** [review]
LGTM!

**Review by naveeddotio — APPROVED** [review]

**Pull request merged**